src/network/C4NetIO.cpp

```diff
@@ -1769,20 +1814,21 @@ bool C4NetIOSimpleUDP::InitBroadcast(addr_t *pBroadcastAddr)
 	}
 
 	// set mc ttl to somewhat about "same net"
-	int iTTL = 16;
-	if (setsockopt(sock, IPPROTO_IP, IP_MULTICAST_TTL, reinterpret_cast<char*>(&iTTL), sizeof(iTTL)) == SOCKET_ERROR)
+	int TTL = 16;
+	if (setsockopt(sock, IPPROTO_IPV6, IPV6_MULTICAST_HOPS, reinterpret_cast<char*>(&TTL), sizeof(TTL)) == SOCKET_ERROR)
 	{
 		SetError("could not set mc ttl", true);
 		return false;
 	}
 
 	// set up multicast group information
 	this->MCAddr = *pBroadcastAddr;
-	MCGrpInfo.imr_multiaddr = static_cast<sockaddr_in*>(&MCAddr)->sin_addr;
-	MCGrpInfo.imr_interface.s_addr = INADDR_ANY;
+	MCGrpInfo.ipv6mr_multiaddr = static_cast<sockaddr_in6>(MCAddr).sin6_addr;
```

Comment:
I don't think we're guaranteed to have an IPv6 address in MCAddr at this point - [C4NetIOUDP](https://github.com/lluchs/openclonk/blob/18f474a6d16c1377759416c899f06ec3586cc799/src/network/C4NetIO.cpp#L2269) sets an IPv4 before calling this code.